shell/platform/android/io/flutter/plugin/mouse/MouseCursorPlugin.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.mouse;
+
+import android.content.Context;
+import android.view.PointerIcon;
+import androidx.annotation.NonNull;
+import io.flutter.embedding.engine.systemchannels.MouseCursorChannel;
+import java.util.HashMap;
+
+/** A mandatory plugin that handles mouse cursor requests. */
+public class MouseCursorPlugin {
+  @NonNull private final MouseCursorViewDelegate mView;
+  @NonNull private final MouseCursorChannel mouseCursorChannel;
+  @NonNull private final Context context;
+
+  public MouseCursorPlugin(
+      @NonNull MouseCursorViewDelegate view,
+      @NonNull MouseCursorChannel mouseCursorChannel,
+      @NonNull Context context) {
+    mView = view;
+    this.context = context;
+
+    this.mouseCursorChannel = mouseCursorChannel;
+    mouseCursorChannel.setMethodHandler(
+      new MouseCursorChannel.MouseCursorMethodHandler() {
+        @Override
+        public void activateSystemCursor(@NonNull String kind) {
+          mView.setPointerIcon(resolveSystemCursor(kind));
+        }
+      });
+  }
+
+  /**
+   * Return a pointer icon object for a system cursor.
+   *
+   * This method guarantees to return a non-null object.
+   */
+  private PointerIcon resolveSystemCursor(@NonNull String kind) {
+    if (MouseCursorPlugin.systemCursorConstants == null) {
```

Comment:
Since resolveSystemCursor is an instance method and cannot be called unless the constructor is called first, we could initialize systemCursorConstants in the constructor and avoid the if null check on every call.

Reply:
My concern is that this map will be expanded to ~70 cursors in the future, which will not be used by most mobile devices (since they don't set cursors at all). Do you still think it's fine to initialize it in the constructor unconditionally?

Reply:
Ahh, in that case, a comment would be plenty to justify this to future readers :)